src/Adapter/Address/CommandHandler/EditCartAddressHandler.php

```diff
@@ -0,0 +1,172 @@
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Address\CommandHandler;
+
+use Cart;
+use PrestaShop\PrestaShop\Adapter\Validate;
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Address\Command\EditCartAddressCommand;
+use PrestaShop\PrestaShop\Core\Domain\Address\Command\EditCustomerAddressCommand;
+use PrestaShop\PrestaShop\Core\Domain\Address\CommandHandler\EditCartAddressHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Address\Exception\AddressConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Address\Exception\CannotUpdateCartAddressException;
+use PrestaShop\PrestaShop\Core\Domain\Address\ValueObject\AddressId;
+use PrestaShop\PrestaShop\Core\Domain\Cart\CartAddressType;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Exception\CartNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Country\Exception\CountryConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\StateConstraintException;
+use PrestaShopException;
+
+/**
+ * EditCartAddressHandler manages an address update, it then updates cart
+ * relation to the newly created address.
+ */
+class EditCartAddressHandler implements EditCartAddressHandlerInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $commandBus;
+
+    /**
+     * @param CommandBusInterface $commandBus
+     */
+    public function __construct(CommandBusInterface $commandBus)
+    {
+        $this->commandBus = $commandBus;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws AddressConstraintException
+     * @throws CannotUpdateCartAddressException
+     * @throws CountryConstraintException
+     * @throws StateConstraintException
+     */
+    public function handle(EditCartAddressCommand $command): AddressId
+    {
+        try {
+            $cart = new Cart($command->getCartId()->getValue());
+            if (!Validate::isLoadedObject($cart) || $command->getCartId()->getValue() !== (int) $cart->id) {
+                throw new CartNotFoundException(sprintf('Cart with id "%s" was not found', $command->getCartId()->getValue()));
+            }
+
+            $addressCommand = $this->createEditAddressCommand($command, $cart);
+            /** @var AddressId $addressId */
+            $addressId = $this->commandBus->handle($addressCommand);
```

Comment:
```suggestion
            $addressId = $this->editCustomerAddressCommandhandler->handle($addressCommand);
```
We should have handlers call other handlers